packages/generate-arg-types/src/types.ts

```diff
@@ -0,0 +1,169 @@
+import { z } from "zod";
+
+// We're trying to have the same API as Storybook's ArgTypes here
+// https://storybook.js.org/docs/react/api/argtypes
+// https://github.com/ComponentDriven/csf/blob/next/src/story.ts#L63
+//
+// And have the same list of controls as Storybook (with some additions)
+// https://storybook.js.org/docs/react/essentials/controls
+
+const dataType = <T extends string>(name: T) =>
+  z.object({ name: z.literal(name), required: z.boolean() });
+
+const controlType = <T extends string>(name: T) =>
+  z.union([z.literal(name), z.object({ type: z.literal(name) })]);
+
+const controlTypeWithOptions = <T extends string>(name: T) =>
+  z.object({ type: z.literal(name), options: z.array(z.string()) });
+
+const common = {
+  /**
+   * Label for the control (may not match the property name)
+   */
+  name: z.string(),
+  description: z.string().nullable(),
```

Comment:
btw why not .optional()?

Reply:
we actually don'tn want nulls, we use it only when we have to because prismal delivers nulls

Reply:
yeah, mistake. will fix

Reply:
Made `name`, `description `, and `defaultValue` all optional. This is in line with Storybook and will safe a lot of space in json